2-Draft-Accepted/RFCNNNN-PSCore6-and-Windows-Modules.md

```diff
@@ -0,0 +1,81 @@
+---
+RFC: RFCNNNN
+Author: Steve Lee
+Status: Draft
+SupercededBy: n/a
+Version: 0.1
+Area: Engine
+Comments Due: 6/30
+Plan to implement: Yes
+---
+
+# Enable Discovery and Importing of Compatible in-box Windows PowerShell modules
+
+The biggest factor preventing existing Windows PowerShell users from moving to PowerShell Core 6 has been cmdlet coverage.
+Specifically being able to use existing cmdlets that ship with Windows.
+The PowerShell team has been working with Windows teams to port their modules so that they work in PowerShell Core 6.
+This RFC addresses the experience for the user when using PowerShell Core 6.1 with next version of Windows 10.
+
+## Motivation
+
+    As a ITPro,
+    I can use existing Windows PowerShell modules in PowerShell Core 6,
+    so that I can start deploying PowerShell Core 6 in my enterprise.
+
+## Specification
+
+The module manifest (since Windows PowerShell 5) may contain a `CompatiblePSEditions` property to indicate if it is compatible with
+Windows PowerShell (value of `Desktop` even on Windows Server) or PowerShell Core (value of `Core`).
+Currently, this property is only documentation and not observed by PowerShell.
+The proposal is to modify PowerShell Core 6.1 to respect this property.
```

Comment:
What is the proposed behaviour for modules that don't have a manifest? I know the RFC is primarily to address *published* modules, but the handling of any `PSModuleInfo` will be affected by this.

I'm thinking that the manifest-default behaviour of assuming `CompatiblePSEditions = @("Desktop")` would be a serious break for simple `.psm1`s. So some choices there are maybe:
* Set `CompatiblePSEditions = null` in the `PSModuleInfo` and require code paths to handle that (and not treat it like the manifest default).
* Set `CompatiblePSEditions = @("Core")`, on the basis that we assume only compatibility with the loading runtime.
* Set `CompatiblePSEditions = @("Desktop", "Core")`, on the basis that `.psm1`s without a manifest are informal enough that we take a "we won't get in your way; it should be easy to do simple things" policy.